libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -542,7 +570,21 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
                     }
                 }
             }
-            return isDone;
+
+            return authFlowFinished;
+        }
+
+        private boolean isNewLoginUrl(Uri uri) {
+            String host = uri.getHost();
+            String path = uri.getQuery();
```

Comment:
We should call it query.
We should also get the path and make sure it is empty based on @bbirman findings about SSO.